Documentation/gettingstarted/dns.rst

```diff
@@ -0,0 +1,203 @@
+    ^C 
+
+    $ kubectl exec -it spaceship -- curl -sL https://api.twitter.com
+    ^C
+
+    $ kubectl exec -it spaceship -- curl -sL deathstar.default.svc.cluster.local/v1
+    {
+	"name": "Death Star",
+        ...
+        ...
+    }
+
+    $ kubectl exec -it mediabot -- curl -sL deathstar.default.svc.cluster.local/v1
+    {   
+        ...
+        ...
+    }
+
+
+Step 6: DNS Policies Using Patterns 
+===================================
+
+The above policy controlled DNS access based on exact match of the DNS domain name. Often it is required to allow access to a subset of domains. A common example is to control access for cloud services such as AWS S3 which have buckets and regions encoded in the DNS domains. For e.g., the S3 bucket URLs can be of the form, ``<bucketname>.s3.<region>.amazonaws.com``. A pattern based rule such as ``*.s3.*.amazonaws.com`` is needed to allow access to all the S3 buckets irrespective of the region.
+
+Let's use the above example and allow ``mediabot`` pods to access any twitter sub-domain i.e. allow a pattern ``*.twitter.com``. We will achieve this by changing the ``toFQDN`` rule to use ``matchPattern`` instead of ``matchName``.
+
+.. literalinclude:: ../../examples/minikube/dns-pattern.yaml
+
+.. parsed-literal::
+
+  $ kubectl apply -f \ |SCM_WEB|\/examples/minikube/dns-pattern.yaml
+
+Test that ``mediabot`` now has access to multiple twitter services for which the DNS matches the pattern ``*.twitter.com``. Important to note and test that this doesn't allow access to ``twitter.com`` because the ``*.`` in the pattern requires one subdomain to be present in the DNS name. You can simply add more ``matchName`` and ``matchPattern`` clauses to extend the access. 
+(`Learn more <http://docs.cilium.io/en/latest/policy/language/?highlight=DNS#dns-based>`_ about specifying DNS rules using patterns and names.)
+
+.. parsed-literal:: 
+  
+   $ kubectl exec -it mediabot -- curl -sL https://help.twitter.com
+   ...
+
+   $ kubectl exec -it mediabot -- curl -sL https://about.twitter.com
+   ... 
+
+   $ kubectl exec -it mediabot -- curl -sL https://twitter.com
+   ^C 
+
+
+Step 7: Combining DNS and Port Rules
+====================================
+
+The DNS-based access can be restricted to a specific port by adding an L4 rules section. Continuing with the example, we will restrict access to twitter services to port ``443``. 
+
+.. literalinclude:: ../../examples/minikube/dns-port.yaml
+
+.. parsed-literal::
+
+  $ kubectl apply -f \ |SCM_WEB|\/examples/minikube/dns-port.yaml
+
+Testing, the access to ``https://help.twitter.com`` will succeed since it access on port 443 but access to ``http://help.twitter.com`` will be blocked since it goes to port 80.
+
+.. parsed-literal:: 
+  
+   $ kubectl exec -it mediabot -- curl -sL https://help.twitter.com
+   ...
+
+   $ kubectl exec -it mediabot -- curl -sL http://help.twitter.com
+   ^C  
+
+Step 8: Combining DNS, Port and API-aware Rules
```

Comment:
This step in the guide is not practical. What is `covalent.link` and why would it be unencrypted? This is the DNS guide, let's focus on DNS and make sure to describe how it can be combined with other policy elements. The shorter and easier to consume the guide, the better.

Reply:
Removed it. Idea was to show the L7 rule combination for a private VPC service which might be unencrypted.